packages/dataviews/src/view-table.js

```diff
@@ -291,54 +316,64 @@ function ViewTable( {
 			)
 	);
 	const shownData = useAsyncList( data );
-	const usedData = deferredRendering ? shownData : data;
+	const usefulData = data.length ? data : shownData;
+	const deferredData = deferredRendering ? shownData : usefulData;
+	const staleData = shownData.length ? shownData : previousData.current;
+	const usedData =
+		isLoading && ! deferredData.length ? staleData : deferredData;
+	previousData.current = usedData;
 	const hasData = !! usedData?.length;
-	if ( isLoading ) {
-		// TODO:Add spinner or progress bar..
-		return (
-			<div className="dataviews-loading">
-				<h3>{ __( 'Loading' ) }</h3>
-			</div>
-		);
-	}
+
 	const sortValues = { asc: 'ascending', desc: 'descending' };
+	const tableNoticeId = useId();
+
 	return (
 		<div className="dataviews-table-view-wrapper">
-			{ hasData && (
-				<table className="dataviews-table-view">
-					<thead>
-						<tr>
-							{ visibleFields.map( ( field ) => (
-								<th
-									key={ field.id }
-									style={ {
-										width: field.width || undefined,
-										minWidth: field.minWidth || undefined,
-										maxWidth: field.maxWidth || undefined,
-									} }
-									data-field-id={ field.id }
-									aria-sort={
-										view.sort?.field === field.id &&
-										sortValues[ view.sort.direction ]
+			<table
+				className="dataviews-table-view"
+				aria-busy={ isLoading || undefined }
+				aria-describedby={ tableNoticeId }
+			>
+				<thead>
+					<tr>
+						{ visibleFields.map( ( field, index ) => (
+							<th
+								key={ field.id }
+								style={ {
+									width: field.width || undefined,
+									minWidth: field.minWidth || undefined,
+									maxWidth: field.maxWidth || undefined,
+								} }
+								data-field-id={ field.id }
+								aria-sort={
+									view.sort?.field === field.id &&
+									sortValues[ view.sort.direction ]
+								}
+								scope="col"
+							>
+								<HeaderMenu
+									ref={ ( node ) =>
+										( headerMenuRefs.current[ field.id ] = {
+											node,
+											fallback:
+												visibleFields[ index - 1 ]?.id,
```

Comment:
Should this fall back to `index + 1` when the current index is 0?

Reply:
Yes, it should probably be something rather than nothing.